lighthouse-core/report/html/renderer/details-renderer.js

```diff
@@ -377,6 +379,36 @@ class DetailsRenderer {
     return element;
   }
 
+  /**
+   * @param {LH.Audit.Details.SourceLocationValue} item
+   * @return {Element|null}
+   * @protected
+   */
+  renderSourceLocation(item) {
+    if (!item.url) {
+      return null;
+    }
+
+    // Lines are never shown as zero-indexed.
+    const line = item.line + 1;
+    const column = item.column;
+
+    let element;
+    if (item.urlProvider === 'network') {
+      element = this.renderTextURL(item.url);
+      this._dom.find('a', element).textContent += `:${line}:${column}`;
+    } else {
+      element = this._renderText(`${item.url}:${line}:${column} (from sourceURL)`);
+    }
+
+    element.classList.add('lh-source-location');
+    element.setAttribute('data-url', item.url);
+    // Should pass `line` to `data-line` as zero-indexed, as DevTools expects that for linkifying.
+    element.setAttribute('data-line', String(item.line));
```

Comment:
i think .toString() is a little favored over using the constructor.

Reply:
fyi this file uses `String()` a few times but never `toString`

Reply:
changed to toString()

note, if a plugin happens to provide a bad value here, it would now error the renderer.

Reply:
Sorry, I was agreeing with you that String() is fine in here since it's the established style

(there's only so much we can do against a plugin that refuses to use our types, though. you don't even need a plugin, you could just save a bad lhr to a gist and load it in the viewer)

Reply:
oops, reverted.